File: abc/abc174/abc174_b.cs
// https://atcoder.jp/contests/abc174/submissions/15640692

using System;
using System.Linq;

class Program
{
    static void Main()
    {
        var input = Console.ReadLine().Split().Select(long.Parse).ToArray();
        var N = input[0];
        var D = input[1];
        var D2 = D * D;

        var result = 0;
        for (int i = 0; i < N; i++)
        {
            var input2 = Console.ReadLine().Split().Select(long.Parse).ToArray();
            var X = input2[0];
            var Y = input2[1];
            if (X * X + Y * Y <= D2)
            {
                result++;
            }
        }
        Console.WriteLine(result);
    }
}

File: abc/abc174/abc174_b.fs
// https://atcoder.jp/contests/abc174/submissions/15640794

let [| N; D |] = stdin.ReadLine().Split() |> Array.map int64
let D2 = D * D

let rec solver result i =
    if i = N then result
    else
        let [| X; Y |] = stdin.ReadLine().Split() |> Array.map int64
        if X * X + Y * Y <= D2 then solver (result + 1) (i + 1L)
        else solver result (i + 1L)

solver 0 0L |> stdout.WriteLine
